File: flutter_app/lib/core/widgets/form_controls.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';

/// TekTech AppCheckbox Component
///
/// Consistent checkbox with label
/// - WCAG AA minimum touch target
/// - Semantic accessibility
class AppCheckbox extends StatelessWidget {
  final bool checked;
  final ValueChanged<bool> onChanged;
  final String label;
  final bool enabled;

  const AppCheckbox({
    super.key,
    required this.checked,
    required this.onChanged,
    required this.label,
    this.enabled = true,
  });

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: enabled ? () => onChanged(!checked) : null,
      child: Padding(
        padding: const EdgeInsets.symmetric(vertical: 8),
        child: Row(
          children: [
            SizedBox(
              width: 48,
              height: 48,
              child: Checkbox(
                value: checked,
                onChanged: enabled
                    ? (value) => onChanged(value ?? false)
                    : null,
                activeColor: AppTheme.primaryColor,
              ),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Text(
                label,
                style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                  color: enabled
                      ? AppTheme.textPrimaryColor
                      : AppTheme.textDisabledColor,
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

/// TekTech AppSwitch Component
///
/// Consistent switch with label and optional description
/// - WCAG AA minimum touch target
/// - Semantic accessibility
class AppSwitch extends StatelessWidget {
  final bool checked;
  final ValueChanged<bool> onChanged;
  final String label;
  final String? description;
  final bool enabled;

  const AppSwitch({
    super.key,
    required this.checked,
    required this.onChanged,
    required this.label,
    this.description,
    this.enabled = true,
  });

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: enabled ? () => onChanged(!checked) : null,
      child: Padding(
        padding: const EdgeInsets.symmetric(
          horizontal: AppTheme.spacing16,
          vertical: 8,
        ),
        child: Row(
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    label,
                    style: Theme.of(context).textTheme.bodyLarge?.copyWith(
                      color: enabled
                          ? AppTheme.textPrimaryColor
                          : AppTheme.textDisabledColor,
                    ),
                  ),
                  if (description != null) ...[
                    const SizedBox(height: 4),
                    Text(
                      description!,
                      style: Theme.of(context).textTheme.bodySmall?.copyWith(
                        color: AppTheme.textSecondaryColor,
                      ),
                    ),
                  ],
                ],
              ),
            ),
            const SizedBox(width: 16),
            SizedBox(
              width: 48,
              height: 48,
              child: Center(
                child: Switch(
                  value: checked,
                  onChanged: enabled ? onChanged : null,
                  activeTrackColor: AppTheme.primaryColor,
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: frontend/android-native/app/src/main/java/com/example/minitasktracker/core/ui/components/FormControls.kt
package com.example.minitasktracker.core.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.role
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.example.minitasktracker.core.ui.theme.Sizes
import com.example.minitasktracker.core.ui.theme.Spacing

/**
 * TekTech AppCheckbox Component
 * 
 * Consistent checkbox with label
 * - WCAG AA minimum touch target
 * - Semantic accessibility
 * 
 * @param checked Whether checkbox is checked
 * @param onCheckedChange Checked state change callback
 * @param label Checkbox label text
 * @param enabled Whether checkbox is enabled
 */
@Composable
fun AppCheckbox(
  checked: Boolean,
  onCheckedChange: (Boolean) -> Unit,
  label: String,
  modifier: Modifier = Modifier,
  enabled: Boolean = true
) {
  Row(
    modifier = modifier
      .semantics { role = Role.Checkbox }
      .defaultMinSize(minHeight = Sizes.minTouchTarget),
    verticalAlignment = Alignment.CenterVertically,
    horizontalArrangement = Arrangement.spacedBy(Spacing.spacing12)
  ) {
    Checkbox(
      checked = checked,
      onCheckedChange = onCheckedChange,
      enabled = enabled,
      colors = CheckboxDefaults.colors(
        checkedColor = MaterialTheme.colorScheme.primary,
        uncheckedColor = MaterialTheme.colorScheme.outline
      )
    )
    Text(
      text = label,
      style = MaterialTheme.typography.bodyMedium,
      color = if (enabled) {
        MaterialTheme.colorScheme.onSurface
      } else {
        MaterialTheme.colorScheme.onSurface.copy(alpha = 0.38f)
      }
    )
  }
}

/**
 * TekTech AppSwitch Component
 * 
 * Consistent switch with label
 * - WCAG AA minimum touch target
 * - Semantic accessibility
 * 
 * @param checked Whether switch is checked
 * @param onCheckedChange Checked state change callback
 * @param label Switch label text
 * @param enabled Whether switch is enabled
 */
@Composable
fun AppSwitch(
  checked: Boolean,
  onCheckedChange: (Boolean) -> Unit,
  label: String,
  modifier: Modifier = Modifier,
  enabled: Boolean = true,
  description: String? = null
) {
  Row(
    modifier = modifier
      .fillMaxWidth()
      .semantics { role = Role.Switch }
      .defaultMinSize(minHeight = Sizes.minTouchTarget)
      .padding(horizontal = Spacing.screenPadding, vertical = Spacing.spacing8),
    horizontalArrangement = Arrangement.SpaceBetween,
    verticalAlignment = Alignment.CenterVertically
  ) {
    Column(
      modifier = Modifier.weight(1f),
      verticalArrangement = Arrangement.spacedBy(Spacing.spacing4)
    ) {
      Text(
        text = label,
        style = MaterialTheme.typography.bodyLarge,
        color = if (enabled) {
          MaterialTheme.colorScheme.onSurface
        } else {
          MaterialTheme.colorScheme.onSurface.copy(alpha = 0.38f)
        }
      )
      description?.let {
        Text(
          text = it,
          style = MaterialTheme.typography.bodySmall,
          color = MaterialTheme.colorScheme.onSurfaceVariant
        )
      }
    }
    
    Spacer(modifier = Modifier.width(Spacing.spacing16))
    
    Switch(
      checked = checked,
      onCheckedChange = onCheckedChange,
      enabled = enabled,
      colors = SwitchDefaults.colors(
        checkedThumbColor = MaterialTheme.colorScheme.primary,
        checkedTrackColor = MaterialTheme.colorScheme.primaryContainer,
        uncheckedThumbColor = MaterialTheme.colorScheme.outline,
        uncheckedTrackColor = MaterialTheme.colorScheme.surfaceVariant
      )
    )
  }
}
